converter/src/test/java/org/dswarm/converter/flow/test/json/JSONSourceResourceGDMStmtsFlowTest.java

```
@@ -0,0 +1,205 @@
+
+		final String dataModelUUID = "1";
+
+		final DataModel dataModel = new DataModel(dataModelUUID);
+
+		dataModel.setConfiguration(configuration);
+
+		final JSONSourceResourceGDMStmtsFlow flow = injector
+				.getInstance(JsonResourceFlowFactory.class)
+				.fromDataModel(dataModel, false);
+
+		testFlow(flow, "bib-record-marc.json", "test-record-marc-gdm.json", null);
+	}
+
+	private void testFlow(final JSONSourceResourceGDMStmtsFlow flow, final String fileName, final String expectedResultFileName, final Integer offset)
+			throws DMPConverterException {
+
+		final List<GDMModel> gdmModels = flow.applyResource(fileName).toList().toBlocking().first();
+
+		LOG.debug("GDM model = '{}'", gdmModels);
+
+		if (gdmModels != null && !gdmModels.isEmpty()) {
+
+			for (final GDMModel gdmModel : gdmModels) {
+
+				final Model model = gdmModel.getModel();
+
+				final JsonNode jsonNode = gdmModel.toJSON();
+
+				Assert.assertNotNull("the GDM model shouldn't be null", model);
+
+				// serialise model as JSON (pretty print)
+				final ObjectMapper mapper = DMPPersistenceUtil.getJSONObjectMapper().copy().configure(SerializationFeature.INDENT_OUTPUT, true);
+
+				String modelJSON = null;
+
+				try {
+
+					modelJSON = mapper.writeValueAsString(model);
+				} catch (final JsonProcessingException e) {
+
+					Assert.assertTrue("something went wrong while serialising the GDM model to JSON", false);
+				}
+
+				Assert.assertNotNull("the GDM model JSON shouldn't be null", modelJSON);
+
+				final URL expectedResultFileURL = Resources.getResource(expectedResultFileName);
+
+				Assert.assertNotNull("the expected result file URL shouldn't be null", expectedResultFileURL);
+
+				String expectedResult = null;
+
+				try {
+
+					expectedResult = Resources.toString(expectedResultFileURL, Charsets.UTF_8);
+				} catch (final IOException e) {
+
+					Assert.assertTrue("something went wrong while reading expected result from file", false);
```

Also, maybe its better to have the test method throw the exception, which will also fail the test but not swallow the reason for it.